### PR TITLE
TMA-892: User filters brick dry run

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 1]
```diff
@@ -385,7 +385,12 @@ def self.execute_mufs(user_filters, options = {})
       else
         GoodData.logger.warn("Data permissions computed: #{to_create.count} to create and #{to_delete.count} to delete")
       end
-      return { created: to_create, deleted: to_delete } if dry_run
+
+      if dry_run
+        create_results = to_create.map { |x| { status: 'dry_run', user: x.first, type: 'create' } }
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'x.first' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has the variable name 'x' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/5/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/816/ : SUCCESS in 41s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1084/ : SUCCESS in 1m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/7/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/817/ : SUCCESS in 39s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1085/ : SUCCESS in 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/8/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/818/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1086/ : SUCCESS in 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/9/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/819/ : SUCCESS in 40s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1087/ : SUCCESS in 49s

**Comment by kubamahnert** [comment]
ok to merge

File: lib/gooddata/connection.rb [hunk 2]
```diff
@@ -118,7 +118,7 @@ def sso_url(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts
     # @param [String] login Email address used for logging into gooddata
     # @param [String] provider Name of SSO provider
     # @return [GoodData::Rest::Client] Instance of REST client
-    def connect_sso(login, provider, url, opts)
+    def connect_sso(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts = {})
```
Comment (yenkins) [inline lib/gooddata/connection.rb]: Has 4 parameters (LongParameterList)

File: lib/gooddata/connection.rb [hunk 3]
```diff
@@ -66,7 +66,7 @@ def with_connection(options = nil, second_options = nil, third_options = {}, &bl
     # @param [Hash] opts Additional options
     # @option opts [Fixnum] :validity Validity in seconds from 'now'
     # @return [String] URL which can be used for SSO logging in
-    def sso_url(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts = DEFAULT_SSO_OPTIONS)
+    def sso_url(login, provider, url, opts = DEFAULT_SSO_OPTIONS)
```
Comment (yenkins) [inline lib/gooddata/connection.rb]: Has 4 parameters (LongParameterList)

Comment (yenkins) [inline lib/gooddata/connection.rb]: Has approx 22 statements (TooManyStatements)

File: lib/gooddata/lcm/lcm2.rb [hunk 4]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
+        # Check if all required parameters were passed
+        BaseAction.check_params(action.const_get('PARAMS'), params)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Calls 'action.const_get('params')' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm2.rb [hunk 5]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Calls 'params.clear_filters' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm2.rb [hunk 6]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Has approx 7 statements (TooManyStatements)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/10/ : SUCCESS in 25s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/821/ : SUCCESS in 3m 26s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1089/ : SUCCESS in 3m 14s

**Comment by panjan** [comment]
Closing in favor of https://github.com/gooddata/gooddata-ruby/pull/1156

**Pull request closed without merging**